### ISSUE TITLE
`gc`: julia `1.6` removes artifacts

### ISSUE THREAD

**Issue by t-bltg** [JuliaLang/Pkg.jl#3159]
I have a centralized depot with artifacts.

Running `Pkg.gc()` removes artifacts in this depot in `1.6` ...

https://github.com/JuliaLang/Pkg.jl/pull/2654 fixes the issue (no removal using `1.7`).

@fredrikekre, based on your comment https://github.com/JuliaLang/Pkg.jl/pull/2654#issuecomment-871279249, would there be arguments against backporting https://github.com/JuliaLang/Pkg.jl/pull/2654 in `1.6` ?

**Comment by fredrikekre** [comment]
It can certainly be backported, but given https://github.com/JuliaLang/Pkg.jl/pull/2654#issuecomment-871279249 I guess I thought it wasn't necessary on 1.6, but you say it is?

**Comment by t-bltg** [comment]
Yes, imo running `gc()` on `1.6` should only remove artifacts in `depots1` by default.

**Comment by KristofferC** [comment]
This should have been resolved now (by 1.10 being the LTS...)
